frontend/src/lib/components/SelectBox.tsx

```diff
@@ -246,25 +260,53 @@ export function SelectUnit({
         }
     }
 
+    const isRowLoaded = ({ index }: { index: number }): boolean => {
+        return Boolean(flattenedData[index])
+    }
+
+    const loadableRowCount = Object.values(loadableItems).reduce(
+        (total, { totalCount: count = 0 }) => (total += count),
+        0
+    )
+    const staticRowCount = Object.values(staticItems).reduce(
+        (total, { dataSource: { length } }) => (total += length),
+        0
+    )
+    const rowCount = loadableRowCount + staticRowCount
```

Comment:
I'll echo an [older commit of mine](https://github.com/PostHog/posthog/pull/4727#discussion_r650778940)

> Let's avoid using `reduce` whenever possible. The only thing using `reduce` shows to whoever is reading the code is that the author is proficient in using `reduce`, _nothing_ else :). It's usually longer and always less readable (`==` maintainable) than all other ways of doing the same thing.

In this case, just += the values in the `Object.entries(items).forEach` loop above.